playbooks/prepare_stack_testconfig.yaml

```diff
@@ -0,0 +1,92 @@
+---
+- hosts: standalone
+  become: true
+  become_user: stack
+  gather_facts: false
+  vars_files: vars/defaults.yaml
+
+  tasks:
+
+  - name: Gather host environment variables
+    setup:
+      filter: ansible_env
+
+  - name: Ensure remote .ssh directory
+    file:
+      path: "{{ ansible_env.HOME }}/.ssh"
+      state: directory
+      owner: stack
+      group: stack
+      mode: '0700'
+
+  - name: Copy ssh public key to host
+    copy:
+      src: "{{ testconfig_public_key }}"
+      dest: "{{ ansible_env.HOME }}/.ssh/{{ testconfig_public_key | basename }}"
+      owner: stack
+      group: stack
+      mode: '0600'
+    register: testconfig_public_key_copy
+
+  - name: Configure resources for openshift user
+    shell: |
+      set -e -o pipefail
+
+      if ! openstack keypair show default >/dev/null; then
+          openstack keypair create --public-key {{ testconfig_public_key_copy.path }} default
+      fi
+      if ! openstack network show private >/dev/null; then
+          openstack network create --internal private
+      fi
+      if ! openstack subnet show private-subnet >/dev/null; then
+          openstack subnet create private-subnet --subnet-range {{ testconfig_private_cidr }} --network private
+      fi
+      # create basic security group to allow ssh/ping/dns
+      if ! openstack security group show basic >/dev/null; then
+          openstack security group create basic
+      fi
+      if ! openstack security group rule list basic | grep "22:22"; then
+          openstack security group rule create basic --protocol tcp --dst-port 22:22 --remote-ip 0.0.0.0/0
+      fi
+      if ! openstack security group rule list basic | grep "icmp"; then
+          openstack security group rule create --protocol icmp basic
+      fi
+      if ! openstack security group rule list basic | grep "53:53"; then
+          openstack security group rule create --protocol udp --dst-port 53:53 basic
```

Comment:
since it's OUTPUT rule, I guess it's already open by default, isn't? I never had to create such a rule, I wonder why you added it here.

Reply:
I just copied this 😂 I have no idea why it's there tbh.

Reply:
I am not sure if i am following the context of where this rules are being created but if Openstck security groups are usually created with source/dest requirements.  And they might not be necessary OUTPUT. 
it could be that thi is a security group to be used bewteen hosts in the same subnet (tenant). 
we do the same in openshift setup because the dns server (port 53) and the workers are in the same subnet, but they are not allowed to talk to each other unless a rule exists in the security group that controls their traffic

Reply:
I copied all this stuff from ocp-doit. We don't actually need it at all and it uses additional VIPs, which is why I moved it into a separate play. The only thing I think this stuff is useful for is if you want to test your openstack by creating VMs manually.

Our team should probably almost never run it. I only didn't delete it because I have used it when testing dev-install.